tests/client/amo/sagas/testCategories.js

```diff
@@ -0,0 +1,70 @@
+import { hideLoading, showLoading } from 'react-redux-loading-bar';
+import { call, put, select, takeEvery } from 'redux-saga/effects';
+
+import categoriesSaga, { fetchCategories } from 'amo/sagas/categories';
+import { getApi } from 'amo/sagas/utils';
+import * as actions from 'core/actions/categories';
+import { categories as categoriesApi } from 'core/api';
+import { CATEGORIES_FETCH } from 'core/constants';
+
+
+const categories = {};
+
+describe('categoriesSaga', () => {
+  it('should get Api from state then make API request to categories', () => {
+    const fetchCategoriesGenerator = fetchCategories();
+
+    let next = fetchCategoriesGenerator.next();
```

Comment:
Oof, I know you are just following the docs but I think this testing approach is very fragile and not very helpful. I agree with all the same problems raised in this issue: https://github.com/redux-saga/redux-saga/issues/518

I did some digging and [this library](https://github.com/wix/redux-saga-tester) looks the most promising as a solution. Do you want to try rewriting the tests with that to see how it feels?